provider/pkg/provider/util.go

```diff
@@ -87,31 +124,31 @@ func parseKubeconfigPropertyValue(kubeconfig resource.PropertyValue) (*clientapi
 		return &clientapi.Config{}, nil
 	}
 
-	var cfg []byte
+	var cfg string
 	if kubeconfig.IsString() {
-		cfg = []byte(kubeconfig.StringValue())
+		cfg = kubeconfig.StringValue()
 	} else if kubeconfig.IsObject() {
 		raw := kubeconfig.ObjectValue().Mappable()
 		jsonBytes, err := json.Marshal(raw)
 		if err != nil {
 			return nil, fmt.Errorf("failed to unmarshal kubeconfig: %v", err)
 		}
-		cfg = jsonBytes
+		cfg = string(jsonBytes)
 	} else {
 		return nil, fmt.Errorf("unexpected kubeconfig format, type: %v", kubeconfig.TypeString())
 	}
-	config, err := clientcmd.Load(cfg)
+	config, err := parseKubeconfigString(cfg)
```

Comment:
This change is the essential one that teaches `DiffConfig` to support a file-based `kubeconfig` property, by reusing the logic of `Configure`.